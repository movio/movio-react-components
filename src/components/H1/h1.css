@import "variables.css";

:root {
  --h1-font-family: var(--font-primary);
  --h1-font-size: 2.15rem;
  --h1-color: var(--color-primary);
}

.h1 {
  font-family: var(--h1-font-family);
  font-size: var(--h1-font-size);
  color: var(--h1-color);
  font-weight: 600;
  text-transform: uppercase;
  letter-spacing: 0.025em;
}
