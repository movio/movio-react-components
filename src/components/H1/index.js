// @flow
import React from 'react';
import classnames from 'classnames';

import styles from './h1.css';

type Props = {
  className?: string,
  children?: ReactChildren,
}

const H1 = ({ className = '', children }: Props) =>
  <h1 className={classnames(className, styles.h1)}>
    {children}
  </h1>;

export default H1;
export {
  styles,
};

