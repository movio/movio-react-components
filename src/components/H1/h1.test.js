import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import H1, { styles } from './index';

describe('<H1 />', () => {
  it('should correctly render children, when text', () => {
    const component = <H1>Title</H1>;
    const wrapper = shallow(component);
    expect(wrapper.text()).toEqual('Title');
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('h1-children-text');
  });

  it('should append a className to the default, if provided', () => {
    const component = <H1 className="test-class">Title H1</H1>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(`test-class ${styles.h1}`);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('h1-custom-classname');
  });

  it('should use the default className from the stylesheet, if not provided', () => {
    const component = <H1>Title H1</H1>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(styles.h1);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('h1-default-classname');
  });
});
