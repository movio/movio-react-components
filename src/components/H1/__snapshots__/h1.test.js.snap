// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h1-children-text 1`] = `
<h1
  className="h1"
>
  Title
</h1>
`;

exports[`h1-custom-classname 1`] = `
<h1
  className="test-class h1"
>
  Title H1
</h1>
`;

exports[`h1-default-classname 1`] = `
<h1
  className="h1"
>
  Title H1
</h1>
`;
